### PR TITLE
Add type-level list of Symbols and implement insertion sort

### PR DESCRIPTION
**Pull request by kcsongor**
A type-level list of symbols with an insertion sort implementation (in ascending order). If this makes sense for the typelevel prelude, then I could add reversing and appending of these lists too.
(I'm not sure if I correctly followed the coding style)

### CODE REVIEW

File: src/Type/Data/Symbol.purs [hunk 1]
```diff
@@ -45,3 +51,41 @@ instance equalsSymbol
...
+-- The actual insertion function, pattern matching on the result of the comparison
+class InsertI (c :: Ordering)
+              (x :: Symbol)
+              (y :: Symbol)
+              (ys :: SList)
+              (o :: SList) |
+              c x y ys -> o
+
+instance insertLT :: InsertI LT x y ys (SCons x (SCons y ys))
+instance insertGT :: InsertI GT x y ys (SCons y (SCons x ys))
```
Comment (LiamGoodacre) [inline src/Type/Data/Symbol.purs]: Don't you need to recurse and rebuild `ys` in the `GT` case?

Comment (kcsongor) [reply src/Type/Data/Symbol.purs]: Ah, indeed!

**Comment by kcsongor** [comment]
Currently, the `Insert` and `InsertI` classes aren't exported - should they be, for the constraint solver to pick up the instances?

**Comment by paf31** [comment]
I do think this is interesting and potentially quite useful, but I'm not sure if it's a match for this library. I don't think we'd put the value level equivalent in `prelude` for example. I assume it go in a separate library easily enough?

**Comment by kcsongor** [comment]
Yeah, that's perfectly reasonable - I'll close this PR then, thanks!

**Pull request closed without merging**